e2e-tests/src/test/finalization.rs

```diff
@@ -10,12 +11,15 @@ pub async fn finalization() -> anyhow::Result<()> {
     let config = setup_test();
     let connection = config.create_root_connection().await;
 
-    let finalized = connection.connection.get_finalized_block_hash().await;
+    let finalized = connection.connection.get_finalized_block_hash().await?;
     let finalized_number = connection
         .connection
         .get_block_number(finalized)
-        .await
-        .unwrap();
+        .await?
+        .ok_or(anyhow!(
+            "Failed to retrieve block number for hash {:?}",
```

Comment:
nit: you can inline `finalized` into the literal, like `anyhow!("Failed to retrieve block number for hash {finalized:?}")`

Reply:
done